cradle-core/src/main/java/com/exactpro/cradle/testevents/TestEventSingleToStore.java

```diff
@@ -70,8 +71,23 @@ public void setSuccess(boolean success)
 		this.success = success;
 	}
 	
-	public void setMessages(Set<StoredMessageId> messages)
+	public void setMessages(Set<StoredMessageId> messages) throws CradleStorageException
 	{
+		validateAttachedMessageIds(messages);
 		this.messages = messages;
 	}
+
+	private void validateAttachedMessageIds(Set<StoredMessageId> ids) throws CradleStorageException
+	{
+		if (ids == null)
+			return;
+		BookId eventBookId = getId().getBookId();
+		for (StoredMessageId id : ids)
+		{
+			BookId messageBookId = id.getBookId();
+			if (!eventBookId.equals(messageBookId))
+				throw new CradleStorageException("Book of message is (" +
+						messageBookId + ") differs from the event book (" + eventBookId + ")");
```

Comment:
throw new CradleStorageException("Book of message (" +
						messageBookId + ") differs from the event book (" + eventBookId + ")");